### ISSUE TITLE
Spot datafeed is being ignored

### ISSUE THREAD

**Issue by thetaiter** [kubecost/cost-analyzer-helm-chart#1210]
**Describe the bug**
Kubecost does not seem to be able to read the data in my spot datafeed S3 bucket. I can confirm that the datafeed is dumping files to the S3 bucket. I can also confirm that Kubecost does have permissions to read the bucket. I can also confirm that we do have active spot instances in our AWS account. However in the settings in Kubecost, it just says `No spot instances detected` forever. In the logs, I see these messages every 15 minutes:
```
I0121 12:17:49.943883       1 awsprovider.go:2207] Found 0 spot data files from yesterday
I0121 12:17:49.956519       1 awsprovider.go:2216] Found 0 spot data files from today
I0121 12:17:49.956581       1 awsprovider.go:920] Spot Pricing Refresh scheduled in 15.00 minutes.
```

**To Reproduce**
1. Set up spot datafeed to write to an S3 bucket in AWS.
2. Setup the AWS values in this helm chart so that Kubecost can read it:
```
kubecostProductConfigs:
  clusterName: "<cluster-name>"
  createServiceKeySecret: true
  projectID: <aws-account-id>
  awsSpotDataRegion: us-west-2
  awsSpotDataBucket: <s3-bucket-name>
  awsServiceKeyName: <aws-acess-key-id>
  awsServiceKeyPassword: <aws-secret-access-key>
```
3. Deploy Kubecost and go to settings to verify that there is a warning that says `No spot instances detected`.
4. Spin up an EKS node group with a spot autoscaling group spanning multiple availability zones.
5. Wait for new a new datafile to be written to the spot datafeed S3 bucket.
6. Wait for Kubecost to refresh (every 15 minutes) spot data.
7. Go to settings in kubecost and it still says `No spot instances detected`.

**Expected behavior**
I expect Kubecost to detect my spot EC2 instances using the data found in the spot datafeed S3 bucket.

**Screenshots**
![image](https://user-images.githubusercontent.com/6549935/150570794-a7dbf098-bb70-4d84-8d8f-6fc9219d3bb9.png)

**Collect logs (please complete the following information):**

 - Run `helm ls` and paste the output here:
```
|> helm3 ls
NAME    	NAMESPACE	REVISION	UPDATED                            	STATUS  	CHART               	APP VERSION
kubecost	kubecost 	17      	2021-05-03 07:54:07.03152 -0500 CDT	deployed	cost-analyzer-1.79.1	1.79.1
```
 - If the pod is stuck in init, run `kubectl logs <kubecost-cost-analyzer pod name> -n kubecost -c cost-analyzer-init` and paste output here:
The pod is not stuck in init
 - If a page is broken or not loading, open your javascript developer console and paste the ouput here:
 - All pages are loading properly

 - Run `kubectl logs <kubecost-cost-analyzer pod name> -n kubecost -c cost-analyzer-server` and paste the output here:
```
2022/01/20 22:17:41 Thanos is disabled for the API endpoints.
2022/01/20 22:17:41 Server is ready. 
```
 - Run `kubectl logs <kubecost-cost-analyzer pod name> -n kubecost -c cost-model` and paste the output here:
[kubecost-cost-model.log](https://github.com/kubecost/cost-analyzer-helm-chart/files/7915520/kubecost-cost-model.log)

 - Run `kubectl logs <kubecost-cost-analyzer pod name> -n kubecost -c cost-analyzer-frontend` and paste the output here:
This contained some internal data and I don't have an easy way to obfuscate it, but I'm not sure this particular log is relevant in this case. If it is, just ask and I can find a way to send some info from this log.

gz#1275

**Comment by kbrwn** [comment]
@thetaiter Can you try updating to the latest release [v1.89.1](https://github.com/kubecost/cost-analyzer-helm-chart/releases/tag/1.89.1) and report back if this issue persists? You should be able to do `helm repo update` then an `helm upgrade kubecost kubecost/cost-analyzer -n kubecost` cmd to get to latest.

**Comment by thetaiter** [comment]
I am on the latest version and the problem still exists. That helm3 release is actually just an old entry that we forgot to clean up, we actually deploy this with ArgoCD.

**Comment by thetaiter** [comment]
I think I found the problem 🤦 I was not quoting our AWS account ID in the helm chart so it was being treated as an integer and got converted to a number in exponential notation. I verified this in the UI. I should have done that before coming here, sorry 😅  I am still having an issue with the `awsSpotDataPrefix` value not being respected. I explicitly set it to a blank string (our spot datafeed spills directly to the top level of the S3 bucket) but it appears to be getting automatically populated with the value of `clusterName`. If I update it to a blank string in the UI,m I start to see spot instanced being detected.
